qt/applications/workbench/workbench/plotting/propertiesdialog.py

```diff
@@ -210,16 +208,27 @@ def __init__(self, canvas, axes):
 
         self.ui.gridBox.hide()
 
-        self.images = self.canvas.figure.gca().images
-        if len(self.images) == 0:
-            self.images = [col for col in self.canvas.figure.gca().collections if isinstance(col, QuadMesh)
-                           or isinstance(col, Poly3DCollection)]
+        self.images=[]
+
+        images = get_images_from_figure(canvas.figure)
+        # If there are an equal number of plots and colorbars so apply changes to plot with the selected colorbar
+        # Otherwise apply changes to all the plots in the figure
+        if len(images) != len(self.canvas.figure.axes)/2:
```

Comment:
Bit confused here, so when does it hit the else statement?

Reply:
I think when I was writing it I was thinking if you had more than one colorbar, so say if you had 2 plots each with their own colorbar. It is possible to do this from a script but I don't think anyone does. I can remove it if you think it's unnecessary.

Reply:
Ah I see, makes sense. I would say its fine to keep it in there. On the off chance that they create their own colorfill plots it’s probably the behaviour they would desire.